File: platform/College-1/Litcoder/Week 4/Code/lab13.py
class node:
    def __init__(self,data):
        self.data = data
        self.next = None

class linked_list:
    def __init__(self):
        self.head = None
    
    def add_to_end(self,data):
        new_node = node(data)
        if self.head==None:
            self.head = new_node
            return
        current_node = self.head
        while current_node.next is not None:
            current_node = current_node.next
        current_node.next = self.head

    def remove_from_beginning(self):
        if self.head==None:
            return
        self.head = self.head.next

    def print(self):
        if self.head == None:
            return
        print(self.head.data)

if __name__ == "__main__":
    list = linked_list()
    input_string = input().split(',')
    for command in input_string:
        try:
            part = command.split(' ')
            if part[0] == '1':
                list.add_to_end(part[1])
            elif part[0] == '2':
                list.remove_from_beginning()
            elif part[0] == '3':
                list.print()
            else:
                # Do nothing
                pass
        except IndexError:
            # Do nothing
            pass

File: platform/College-1/Litcoder/Week 4/Code/lab13.java
import java.util.Scanner;

class Node {
    String data;
    Node next;

    Node(String data) {
        this.data = data;
        this.next = null;
    }
}

public class Main {
    private Node head;

    public Main() {
        head = null;
    }

    private void addToEnd(String data) {
        Node newNode = new Node(data);
        if (head == null) {
            head = newNode;
            return;
        }

        Node currentNode = head;
        while (currentNode.next != null) {
            currentNode = currentNode.next;
        }
        currentNode.next = newNode;
    }

    private void removeFromFront() {
        if (head == null) {
            return;
        } else if (head.next == null) {
            head = null;
            return;
        } else {
            head = head.next;
        }
    }

    private void print() {
        if (head == null) {
            return;
        }
        Node current = head; // Use a temporary variable for clarity
        while (current != null) {
            System.out.println(current.data);
            current = current.next;
        }
    }

    public static void main(String[] args) {
        Main list = new Main();
        Scanner in = new Scanner(System.in);
        String[] input = in.nextLine().split(",");

        for (int iter = 0; iter < input.length; iter++) {
            String[] command = input[iter].split(" ");

            try {
                switch (command[0]) {
                    case "1":
                        list.addToEnd(command[1]);
                        break;
                    case "2":
                        list.removeFromFront();
                        break;
                    case "3":
                        list.print();
                        break;
                    // Ignore all other inputs.
                }
            } catch (Exception e) {
                // Ignore ERROR: IndexOutOfBoundsException
            }
        }
    }
}
